src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala

```diff
@@ -886,19 +899,27 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with Stash {
 
   /**
     * Calculates the duration of the refuel session, the provided energy and throws corresponding events
-    *
-    * @param currentTick
-    * @param vehicle
+    * !!!!! This is temporary and needs to be deleted
+    * @param currentTick current time
+    * @param vehicle vehicle to end charging
+    * @param chargingDuration the duration of charging session
+    * @param energyInJoules the energy in joules
     */
-  def handleEndCharging(currentTick: Int, vehicle: BeamVehicle): Unit = {
-
-    val (chargingDuration, energyInJoules) =
-      vehicle.refuelingSessionDurationAndEnergyInJoules(Some(currentTick - vehicle.getChargerConnectedTick()))
-
-    log.debug("Ending refuel session for {} in tick {}. Provided {} J.", vehicle.id, currentTick, energyInJoules)
-    vehicle.addFuel(energyInJoules)
```

Comment:
Same as the other comments in this file - should this be deleted - it has some changes already...so maybe just finish removing it?

Reply:
Same response